### PR TITLE
RESTBase methods json_as* supports HTTP Status Code set

### PR DESCRIPTION
**Pull request by thegabriele97**

### CODE REVIEW

**Review by eueu1997 — APPROVED** [review]

File: src/common/RESTBase.py [hunk 1]
```diff
@@ -9,34 +9,33 @@ class RESTBase:
...
-    def asjson_info(self, data):
+    def asjson_info(self, data, statuscode: int = 200):
+        cherrypy.response.status = statuscode
         return {"info": data}
 
     @cherrypy.tools.json_out()
     def GET(self, *path, **args):
-        cherrypy.response.status = 400
-        return self.asjson_error("bad request")
+        return self.asjson_error("bad request", 405)
```
Comment (giovannipollo) [inline src/common/RESTBase.py:26]: In the old version, you had `cherrypy.response.status = 400`.
Why did you change to 405? I think it should be like this:
```suggestion
        return self.asjson_error("bad request", 400)
```

Same is for the other lines of code where you have 405

Comment (thegabriele97) [reply src/common/RESTBase.py:26]: Because I think it's better. 405 stands for "method not allowed" and this is the case, the method is not allowed if not explicitly implemented

Comment (giovannipollo) [reply src/common/RESTBase.py:26]: Ok good, thanks for the explanation! Now I understand why you did this change

**Review by giovannipollo — APPROVED** [review]

**Review by francescasilvano — APPROVED** [review]

**Comment by giovannipollo** [comment]
Can we merge this branch?
